watchtower/lookout/lookout.go

```diff
@@ -102,6 +103,52 @@ func (l *Lookout) Stop() error {
 	return nil
 }
 
+// fetchBlockWithRetries attempts to fetch a block from the blockchain using
+// its hash.
+// It retries the fetching process if the initial attempt(s) fail, up to a
+// specified maximum number of retries.
+//
+// Parameters:
+// fetcher - An interface of type BlockFetcher, which provides the GetBlock
+// method to fetch the block.
+// hash - A pointer to a chainhash.Hash, representing the hash of the
+// block to be fetched.
+// maxRetries - An integer specifying the maximum number of attempts to
+// fetch the block.
+//
+// The function loops until it either successfully fetches the block or
+// exhausts the maximum number of retries.
+// If a fetch attempt fails (indicated by an error returned
+// from fetcher.GetBlock),
+// the function logs the retry attempt and then tries again,
+// up to maxRetries times.
+//
+// Returns:
+// *wire.MsgBlock - A pointer to the fetched block, or nil if the block could
+// not be fetched after maxRetries attempts.
+// error - An error object indicating the failure reason in case the function
+// is unable to fetch the block; nil if the fetch is successful.
+//
+// Note: The function logs a message before each retry attempt after the first
+// failure.
```

Comment:
I think you can remove all of this :) the first few lines explain it pretty well :)